src/dune_project.ml

```diff
@@ -569,6 +572,7 @@ let anonymous = lazy (
   ; generate_opam_files = false
   ; file_key
   ; dialects = Dialect.DB.builtin
+  ; explicit_js_mode = false
```

Comment:
Could you change this to `true` for >= 2.0? Similarly to how name mangling and implicit transitive deps are being handled. Also, let's add this to the 2.0 change log as well.

Reply:
Fixed. Just to confirm, if this feature goes directly into 2.0, then we won't even expose the switch `explicit_js_mode`, right?

Reply:
IIRC removing this switch would make our code quite cleaner without breaking too much stuff downstream. @diml do you remember what was the plan here originally?